### DISCUSSION TITLE
feat: 流水线TASK执行前支持自动暂停，在暂停前支持编写插件前置处理逻辑

### DISCUSSION THREAD

**Discussion by liyuefeilong** [Tencent/bk-ci#4276] [Ideas]
What would you like to be added:
对于流水线TASK执行前支持自动暂停的插件，希望能支持编写插件前置处理逻辑，这段逻辑在插件暂停之前运行。

Why is this needed:

现有的执行前自动暂停插件，在暂停之前是不会做逻辑处理的。
现在有使用场景是：用户在打开插件进行交互时，执行前自动暂停的插件需要展示一些经过预处理的数据，希望在暂停前可以运行一段前置处理逻辑。

**Comment by zanyzhao** [comment]
这个过于定制化了哈，我们更希望插件的开发流程轻量化，不要过多入侵流水线引擎的执行层面
